src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java

```diff
@@ -0,0 +1,221 @@
+        }
+        return promise.future();
+      });
+  }
+
+  private Promise<Void> moveFromTempToPermanentTable(
+      Function<Tx<List<Transaction>>, Future<Tx<List<Transaction>>>> processTransactions, List<Transaction> transactions) {
+
+    Promise<Void> promise = Promise.promise();
+    Tx<List<Transaction>> tx = new Tx<>(transactions, getPostgresClient());
+
+    startTx(tx).compose(processTransactions)
+      .compose(this::deleteTempTransactions)
+      .compose(HelperUtils::endTx)
+      .setHandler(result -> {
+        if (result.failed()) {
+          HttpStatusException cause = (HttpStatusException) result.cause();
+          log.error("Transactions {} or associated data failed to be processed", cause, tx.getEntity());
+
+          rollbackTransaction(tx).setHandler(res -> promise.fail(cause));
+        } else {
+          log.info("Transactions {} and associated data were successfully processed", tx.getEntity());
+          promise.complete();
+        }
+      });
+
+    return promise;
+  }
+
+  private Future<Transaction> createTempTransaction(Transaction transaction) {
+    Promise<Transaction> promise = Promise.promise();
+
+    if (transaction.getId() == null) {
+      transaction.setId(UUID.randomUUID()
+        .toString());
+    }
+
+    log.debug("Creating new transaction with id={}", transaction.getId());
+
+    getPostgresClient().save(temporaryTransactionTable, transaction.getId(), transaction, reply -> {
+      if (reply.failed()) {
+        log.error("Transaction creation with id={} failed", reply.cause(), transaction.getId());
+        handleFailure(promise, reply);
+      } else {
+        log.debug("New transaction with id={} successfully created", transaction.getId());
+        promise.complete(transaction);
+      }
+    });
+    return promise.future();
+  }
+
+  private Future<JsonObject> getSummary(Transaction transaction) {
+    Promise<JsonObject> promise = Promise.promise();
+
+    log.debug("Get summary={}", getId(transaction));
+
+    getPostgresClient().getById(summaryTable, getId(transaction), reply -> {
+      if (reply.failed()) {
+        log.error("Summary retrieval with id={} failed", reply.cause(), transaction.getId());
+        handleFailure(promise, reply);
+      } else {
+        log.debug("Summary with id={} successfully extracted", transaction.getId());
+        promise.complete(reply.result());
+      }
+    });
+    return promise.future();
+  }
+
+  private Future<List<Transaction>> getTempTransactions(JsonObject summary) {
+    Promise<List<Transaction>> promise = Promise.promise();
+
+    Criterion criterion = getCriterion(summary.getString("id"));
+
+    getPostgresClient().get(temporaryTransactionTable, Transaction.class, criterion, true, false, reply -> {
+      if (reply.failed()) {
+        log.error("Failed to extract temporary transaction by summary id={}", reply.cause(), summary.getString("id"));
```

Comment:
should be sourcePurchaseOrderId instead of summary?

Reply:
sourcePurchaseOrderId is valid only for Encumbrance type, but for Payments and Credits it should be sourceInvoiceInvoiceId.
I haven't found the proper generic name yet.